### PR TITLE
FIX: Restore SignalRO

### LINKED ISSUE #780: SignalRO - Compatibility Broken
The changes added with https://github.com/bluesky/ophyd/commit/7e8d56b71122f2d8a12399f37824be0326a2e666 ended up breaking code at Typhon where `SignalRO` is being used.

### PR DESCRIPTION
**Pull request by hhslepicka**
This PR restores SignalRO which was being heavily used by Typhon for tests.
Closes #780 

Attn. @klauer

### CODE REVIEW

**Comment by hhslepicka** [comment]
@mrakitin nope... We need just the most simple signal, no fancy generation of value via function... 
SignalRO basically received the value and that was it.

**Comment by hhslepicka** [comment]
@mrakitin no idea.

**Comment by mrakitin** [comment]
@hhslepicka, maybe it's a good chance to move it to the "proper" location? The read-only version of a generic `Signal` seems to be useful outside of the simulation framework.

Do you mind to move the function before this PR gets merged? Would it be much work to change the imports in typhon?

**Comment by hhslepicka** [comment]
@mrakitin not much work just a bunch of try/except for the ImportErrors to keep backwards compatibility.

**Comment by danielballan** [comment]
I put a PR against your PR to suggest another possibility: _define_ in it `ophyd.signal` to keep things organized but expose it in `ophyd.sim` to un-break PyDM without imposing a bunch of import updates. https://github.com/hhslepicka/ophyd/pull/1

**Comment by hhslepicka** [comment]
Merged... it looks perfect @danielballan . 
Thanks for looking into it...

File: ophyd/signal.py [hunk 1]
```diff
@@ -397,6 +397,21 @@ def _run_metadata_callbacks(self):
...
+        self._metadata.update(
+            connected=True,
+            write_access=False,
+        )
+
+    def put(self, value, *, timestamp=None, force=False):
+        raise ReadOnlyError("The signal {} is readonly.".format(self.name))
+
+    def set(self, value, *, timestamp=None, force=False):
+        raise ReadOnlyError("The signal {} is readonly".format(self.name))
```
Comment (mrakitin) [inline ophyd/signal.py]:
```suggestion
        raise ReadOnlyError("The signal {} is readonly.".format(self.name))
```

**Comment by mrakitin** [comment]
Great solution, @danielballan! I added a minor punctuation suggestion to have a consistent error message across the methods.

File: ophyd/signal.py (hunk 1 above)
Comment (hhslepicka) [reply ophyd/signal.py]: Addressed with latest commit.

Comment (mrakitin) [reply ophyd/signal.py]: Thanks!

**Comment by hhslepicka** [comment]
I don't have commit privileges here...

File: ophyd/signal.py [hunk 2]
```diff
@@ -397,6 +397,21 @@ def _run_metadata_callbacks(self):
...
+        self._metadata.update(
+            connected=True,
+            write_access=False,
+        )
+
+    def put(self, value, *, timestamp=None, force=False):
+        raise ReadOnlyError("The signal {} is readonly.".format(self.name))
+
+    def set(self, value, *, timestamp=None, force=False):
+        raise ReadOnlyError("The signal {} is readonly.".format(self.name))
```
Comment (mrakitin) [inline ophyd/signal.py]: Can it be implemented as placeholders, something like the snippet below?
```py
    def put(self, *args, **kwargs):
         raise ReadOnlyError("The signal {} is readonly.".format(self.name))

    def set(self, *args, **kwargs):
        self.put(*args, **kwargs)
```

This way one does not need to sync error messages in multiple places. @hhslepicka, @danielballan, @klauer?

Comment (klauer) [reply ophyd/signal.py]: `set = put`

Comment (mrakitin) [reply ophyd/signal.py]: That works for me too.

Comment (hhslepicka) [reply ophyd/signal.py]: I will just merge it now and we can discuss that later on...

Comment (mrakitin) [reply ophyd/signal.py]: Sounds good.

**Pull request merged**